src/NuGet.Clients/PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -211,21 +218,39 @@ private void LoadItems(PackageItemListViewModel selectedPackageItem)
             token.ThrowIfCancellationRequested();
         }
 
-        private async Task<IEnumerable<PackageItemListViewModel>> LoadNextPageAsync(IItemLoader<PackageItemListViewModel> currentLoader, CancellationToken token)
+        private async Task<IEnumerable<PackageItemListViewModel>> LoadNextPageAsync(PackageItemLoader currentLoader, CancellationToken token)
         {
             var progress = new Progress<IItemLoaderState>(
                 s => HandleItemLoaderStateChange(currentLoader, s));
 
-            // trigger loading
-            await currentLoader.LoadNextAsync(progress, token);
+            // if searchResultTask is in progress then just wait for it to complete
+            // without creating new load task
+            if (_initialSearchResultTask != null)
```

Comment:
It will always be called on UI thread, and there will always be single loading operation at a time. So we're good from thread-safety perspective.